code/obj/item/cigarette.dm

```diff
@@ -569,39 +576,55 @@
 
 /obj/item/cigpacket/proc/update_icon()
 	src.overlays = null
-	if (src.cigcount <= 0)
+	if (length(src.cigs) == 0)
 		src.icon_state = "[src.package_style]0"
 		src.desc = "There aren't any cigs left, shit!"
 	else
 		src.icon_state = "[src.package_style]o"
-		src.overlays += "cig[src.cigcount]"
+		src.overlays += "cig[length(src.cigs)]"
 	return
 
 /obj/item/cigpacket/attack_hand(mob/user as mob)
 	if (user.find_in_hand(src))//r_hand == src || user.l_hand == src)
-		if (src.cigcount == 0)
+		if (length(src.cigs) == 0)
 			user.show_text("You're out of cigs, shit! How you gonna get through the rest of the day?", "red")
 			return
 		else
-			var/obj/item/clothing/mask/cigarette/W = new src.cigtype(user)
+			var/obj/item/clothing/mask/cigarette/W = src.cigs[1]
 			user.put_in_hand_or_drop(W)
-			if (src.cigcount != -1)
-				src.cigcount--
+			src.cigs -= W
 		src.update_icon()
 	else
 		return ..()
 	return
 
+/obj/item/cigpacket/attackby(obj/item/W as obj, mob/user as mob)
+	if (istype(W, /obj/item/clothing/mask/cigarette))
+		var/obj/item/clothing/mask/cigarette/cig = W
+		if (cig.on)
+			user.show_text("You can't put a lit cig back in the packet, are you crazy?", "red")
+			return
+		if (length(src.cigs) < src.max_cigs)
+			src.cigs += cig
+			user.u_equip(cig)
```

Comment:
I think this should have something like `cig.set_loc(src)`. Currently it probably drops the cigarette on the ground or worse leaves it embedded inside the person.

Reply:
this has been done.